src/compiler/parser.ts

```diff
@@ -1227,7 +1226,9 @@ namespace ts {
 
         function parsePropertyNameWorker(allowComputedPropertyNames: boolean): PropertyName {
             if (token() === SyntaxKind.StringLiteral || token() === SyntaxKind.NumericLiteral) {
-                return <StringLiteral | NumericLiteral>parseLiteralNode(/*internName*/ true);
+                const node = <StringLiteral | NumericLiteral>parseLiteralNode();
+                internIdentifier(node.text);

```

Comment:
`internIdentifier` won't have the desired effect if used as a statement; I like the way it was before.